src/otg/datasource/finngen/finngen_finemapping.py

```diff
@@ -0,0 +1,234 @@
+                .when(f.col("credibleSetIndex") == 5, f.col("alpha_5"))
+                .when(f.col("credibleSetIndex") == 6, f.col("alpha_6"))
+                .when(f.col("credibleSetIndex") == 7, f.col("alpha_7"))
+                .when(f.col("credibleSetIndex") == 8, f.col("alpha_8"))
+                .when(f.col("credibleSetIndex") == 9, f.col("alpha_9"))
+                .when(f.col("credibleSetIndex") == 10, f.col("alpha_10")),
+            )
+            .drop(
+                "alpha_1",
+                "alpha_2",
+                "alpha_3",
+                "alpha_4",
+                "alpha_5",
+                "alpha_6",
+                "alpha_7",
+                "alpha_8",
+                "alpha_9",
+                "alpha_10",
+            )
+            .withColumn(
+                "logBF",
+                f.when(f.col("credibleSetIndex") == 1, f.col("lbf_1"))
+                .when(f.col("credibleSetIndex") == 2, f.col("lbf_2"))
+                .when(f.col("credibleSetIndex") == 3, f.col("lbf_3"))
+                .when(f.col("credibleSetIndex") == 4, f.col("lbf_4"))
+                .when(f.col("credibleSetIndex") == 5, f.col("lbf_5"))
+                .when(f.col("credibleSetIndex") == 6, f.col("lbf_6"))
+                .when(f.col("credibleSetIndex") == 7, f.col("lbf_7"))
+                .when(f.col("credibleSetIndex") == 8, f.col("lbf_8"))
+                .when(f.col("credibleSetIndex") == 9, f.col("lbf_9"))
+                .when(f.col("credibleSetIndex") == 10, f.col("lbf_10")),
+            )
+            .drop(
+                "lbf_1",
+                "lbf_2",
+                "lbf_3",
+                "lbf_4",
+                "lbf_5",
+                "lbf_6",
+                "lbf_7",
+                "lbf_8",
+                "lbf_9",
+                "lbf_10",
+            )
+        )
+
+        # drop credible sets where logbf < 2. Except when there's only one credible set in region:
+        # 0.8685889638065036 corresponds to np.log10(np.exp(2)), to match the orginal threshold in publication.
+        finngen_finemapping_summaries_df = (
+            spark.read.option("delimiter", "\t")
+            .csv(finngen_finemapping_summaries, header=True)
+            .filter((f.col("cs_log10bf") > 0.8685889638065036) | (f.col("cs") == 1))
+            .withColumn("studyId", f.concat(f.lit("finngen_r9_"), f.col("trait")))
+            .select(
+                "studyId",
+                "region",
+                f.col("cs").cast("integer").alias("credibleSetIndex"),
+                f.col("cs_log10bf").cast("double").alias("credibleSetlog10BF"),
+            )
+        )
+
+        processed_finngen_finemapping_df = processed_finngen_finemapping_df.join(
+            finngen_finemapping_summaries_df,
+            on=["studyId", "region", "credibleSetIndex"],
+            how="inner",
+        )
+
+        toploci_df = (
+            processed_finngen_finemapping_df.withColumn(
```

Comment:
We have a method to extract the top row in a window. See `get_record_with_maximum_value`